### PR TITLE
[relay-runtime] Split files to better maintenance

### PR DESCRIPTION
**Pull request by renanmav**
I've tried to match typings with source code of [relay-runtime](https://github.com/facebook/relay/tree/master/packages/relay-runtime).

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/relay-runtime/index.d.ts [hunk 1]
```diff
@@ -46,880 +46,157 @@ import {
...
-    uploadables?: UploadableMap;
-    onCompleted?:
-        | ((response: TOperation['response'], errors: ReadonlyArray<PayloadError> | null | undefined) => void)
-        | null;
-    onError?: ((error: Error) => void) | null;
-    optimisticUpdater?: SelectorStoreUpdater | null;
-    optimisticResponse?: TOperation['response'] | null;
-    updater?: SelectorStoreUpdater<TOperation['response']> | null;
-}
+// ./lib/mutations/RelayDeclarativeMutationConfig
```
Comment (alloy) [inline types/relay-runtime/index.d.ts]: There’s no more need to have these comments, as that’s self-documenting now.

**Review by alloy — COMMENTED** [review]
This is looking great, thanks!

The CI issues are legit and it’s probably a good idea to test this out on some code-based before merging.

/cc @kastermester @sibelius @voxmatt

**Comment by renanmav** [comment]
I'm getting a problem with Handlers. In `relay-runtime-tests.tsx`:
```
Type '(handle: any) => typeof RelayDefaultHandlerProvider' is not assignable to type 'HandlerProvider'.
````

Seems that I need to Handler to be generic

File: types/relay-runtime/index.d.ts (hunk 1 above)
Comment (renanmav) [reply types/relay-runtime/index.d.ts]: Removed. Follow the exports of source code of [relay-runtime](https://github.com/facebook/relay/blob/master/packages/relay-runtime/index.js)

**Comment by renanmav** [comment]
I'm testing with my codebase

**Comment by renanmav** [comment]
Well, everything seems to be working with my relay setup on [relayable](https://github.com/yottadev/relayable) (my current project).

Just a question. `react-relay@6.x` uses the latest version of `relay-runtime@6.x`? As `react-relay` depends on `relay-runtime`. 

 If so, I think we are ready to merge.

File: types/relay-runtime/lib/handlers/connection/RelayConnectionHandler.d.ts [hunk 2]
```diff
@@ -14,29 +13,22 @@ export interface ConnectionMetadata {
...
 
-export function insertEdgeAfter(record: RecordProxy, newEdge: RecordProxy, cursor?: string | null): void;
+    insertEdgeAfter(record: RecordProxy, newEdge: RecordProxy, cursor?: string | null): void;
 
-export function insertEdgeBefore(record: RecordProxy, newEdge: RecordProxy, cursor?: string | null): void;
+    insertEdgeBefore(record: RecordProxy, newEdge: RecordProxy, cursor?: string | null): void;
 
-export function update(store: RecordSourceProxy, payload: HandleFieldPayload): void;
+    update(store: RecordSourceProxy, payload: HandleFieldPayload): void;
+};
```
Comment (alloy) [inline types/relay-runtime/lib/handlers/connection/RelayConnectionHandler.d.ts]: These functions should not be exported as part of an interface, but rather as separate functions, as was done before. See how they are exported individually here? https://github.com/facebook/relay/blob/7253d354d31786488e6446dee10516fe545c9454/packages/relay-runtime/handlers/connection/RelayConnectionHandler.js#L538-L546

Comment (alloy) [reply types/relay-runtime/lib/handlers/connection/RelayConnectionHandler.d.ts]: Sent you a PR https://github.com/renanmav/DefinitelyTyped/pull/1

**Comment by alloy** [comment]
After my fix, this seems to work well on 3 codebases I’ve tried ✨ 

I’ll ask in the #relay slack channel if anybody else can confirm as well.

File: types/relay-runtime/lib/handlers/connection/RelayConnectionHandler.d.ts (hunk 2 above)
Comment (alloy) [reply types/relay-runtime/lib/handlers/connection/RelayConnectionHandler.d.ts]: Perhaps this needs some test coverage, seeing as it’s an odd one out. Care to add some?

**Review by alloy — CHANGES_REQUESTED** [review]
Marking this as “request changes” solely so it doesn’t get merged unwittingly.

**Comment by renanmav** [comment]
Merged.

Waiting for more tests.

cc @sibelius

**Review by sibelius — APPROVED** [review]

**Review by alloy — APPROVED** [review]

**Comment by alloy** [comment]
Alright, with a second approval in, let’s do this!

**Pull request merged**